### DISCUSSION TITLE
[Solved] Connection on html5 export

### DISCUSSION THREAD

**Discussion by ddragon15** [Marzin-bot/PostgreSQLClient#41] [Q&A] [answered]
getting some errors on export about 'get_status() on nil'
is there a way around it? I would like to establish a direct connection instead of using a restAPI on my server.

![grafik](https://user-images.githubusercontent.com/21277566/155509495-c6fad2ad-1967-4963-86a6-1cd4b29b8d7b.png)

looks like it is related to StreemPeer not accessible on html5 export 
![grafik](https://user-images.githubusercontent.com/21277566/155510624-d115d585-66f5-4b7a-ac52-8f4d2efadc19.png)
source: [Godot Documentation](https://docs.godotengine.org/en/stable/tutorials/export/exporting_for_web.html)

![grafik](https://user-images.githubusercontent.com/21277566/155510764-a3868a80-dc75-4866-bd1a-ed42a9567ec3.png)

awesome plugin, many thanks 👍

**Comment by KuwazyStudio** [comment, marked as answer]
Unfortunately there is no way to use TCP on the web browser. This is an independent limitation of Godot for security reasons. The only protocols available on browser are http, websocket (with or without ssl/tls layer) and others based on WebRTC theology. The Postgres client-server protocol is itself based on TCP and there is no version based on http for example. Postgres will not work on platforms where the TCP protocol is not exposed. So I can't do anything about it 😩. 

In general, it is not up to the players to connect to the postgres server but to the game server to execute the requests instead of the players (for security and performance reasons). Because there is a connection limit to the postgres server and even if you can increase the limit it is bad practice unless you know what you are doing.

If you have other questions, do not hesitate ! Thank you and happy coding!

Marzin Samuel.
